org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -0,0 +1,144 @@
+package org.code.neighborhood;
+
+public class Painter {
+    private int xLocation;
+    private int yLocation;
+    private Direction direction;
+    private int remainingPaint;
+    private Grid grid;
+    private String id;
+
+    public Painter(int x, int y, String direction, int paint) {
+        this.xLocation = x;
+        this.yLocation = y;
+        this.direction = new Direction(direction);
+        this.remainingPaint = paint;
+        this.grid = Grid.getInstance();
+        this.id = "Painter-"+this.grid.registerPainter();
+
+        System.out.println("created painter");
+    }
+
+    // Turn the painter one compass direction left (i.e. North -> West)
+    public void turnLeft() {
+        this.direction.turnLeft();
+    }
+
+    // Move the painter one square forward in the direction the painter
+    // is facing
+    public void move() {
+        if (this.validMovement(this.direction)) {
+            if (this.direction.isNorth()) {
+                this.yLocation++;
+            } else if (this.direction.isSouth()) {
+                this.yLocation--;
+            } else if (this.direction.isEast()) {
+                this.xLocation++;
+            } else {
+                this.xLocation--;
+            }
+        } else {
+            System.out.println("You can't go that way");
+        }
+        System.out.println("New (x,y) : (" + this.xLocation + "," + this.yLocation + ")");
+    }
+
+    // Add paint of the given color to the grid at the location where the
+    // painter currently is
+    public void paint(String color) {
+        this.grid.getSquare(this.xLocation, this.yLocation).setColor(color);
+    }
+
+    // removes all paint on the square where the painter is standing
+    public void scrapePaint() {
+        this.grid.getSquare(this.xLocation, this.yLocation).removePaint();
+    }
+
+    // Returns how many units of paint are in the painter's personal
+    // bucket
+    public int getMyPaint() {
+        return this.remainingPaint;
+    }
+
+    // Hides the painter on the screen
+    public void hidePainter() {
+        System.out.println("You hid the painter");
+    }
+
+    // Shows the painter on the screen
+    public void showPainter() {
+        System.out.println("You displayed the painter");
+    }
+
+    // Painter adds a single unit of paint to their personal bucket
+    // Counter on the bucket on the screen goes down. If the painter
+    // is not standing on a paint bucket, nothing happens
+    public void takePaint() {
+        if (this.grid.getSquare(this.xLocation, this.yLocation).containsPaint()) {
+            this.grid.getSquare(this.xLocation, this.yLocation).collectPaint();
+            this.remainingPaint++;
+        } else {
+            System.out.println("There is no paint to collect here");
+        }
+    }
+
+    // Returns True if there is paint in the square where the painter
+    // is standing.
+    public Boolean isOnPaint() {
```

Comment:
I'm finding the method names very confusing.  A square can contain a bucket of paint or be painted a certain color.  The painter can also paint a square.  All three of these are referred to using the word 'paint' in different contexts.  I think it would be helpful to use different words consistently for each of these three things.  (This could make a big difference for students with less fluency in English.)

Reply:
@hanwalden ^

Reply:
I don't have any suggestions around here. We tried to be clear with the naming, but I can see how it's challenging. 
isOnPaint - standing on Paint
hasPaint - painter has paint in their bucket
paint - action verb